### DISCUSSION TITLE
What does the `__afl_area_ptr` variable do?

### DISCUSSION THREAD

**Discussion by lahiri-phdworks** [AFLplusplus/AFLplusplus#1481] [Q&A]
I need a way to manipulate the coverage based on certain instructions or events in the example that is being fuzzed. How can I make use of `__afl_area_ptr` to do this? Is there a way to do a custom update to the coverage that AFL maintains programatically?

**Comment by domenukk** [comment]
You can use the `__afl_coverage_interesting` function to mark regions in your code.
See this discussion for more functions
https://github.com/AFLplusplus/AFLplusplus/issues/675

**Reply by lahiri-phdworks** [reply]
Thanks for pointing it out. I have used the `__afl_coverage_interesting` function but want to manipulate the raw coverage information. Is to possible to add a function like `__afl_update_coverage(...)`?

**Comment by domenukk** [comment]
I guess you are free to directly manipulate the map, but you'll have to understand what you are doing. What would you want to work differently in an update_coverage function?
